src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementBackedDocumentsService.java

```diff
@@ -79,18 +90,27 @@ private Claim getClaimByExternalId(String externalId, String authorisation) {
         return claimService.getClaimByExternalId(externalId, authorisation);
     }
 
-    @SuppressWarnings("squid:S3655")
-    private byte[] downloadOrGenerateAndUpload(Claim claim, Supplier<byte[]> documentSupplier, String authorisation) {
-        if (claim.getSealedClaimDocument().isPresent()) {
-            URI documentSelfPath = claim.getSealedClaimDocument().get();
-            return documentManagementService.downloadDocument(authorisation, documentSelfPath);
-        } else {
-            PDF document = new PDF(buildSealedClaimFileBaseName(claim.getReferenceNumber()), documentSupplier.get());
+    private DocumentDetails uploadToDocumentManagement(byte[] documentBytes,
+                                         String authorisation,
+                                         String baseFileName) {
+        PDF document = new PDF(baseFileName, documentBytes);
+        URI documentSelfPath = documentManagementService.uploadDocument(authorisation, document);
+        return new DocumentDetails() {
+            @Override
+            public URI getDocumentSelfPath() {
+                return documentSelfPath;
+            }
 
-            URI documentSelfPath = documentManagementService.uploadDocument(authorisation, document);
-            claimService.linkSealedClaimDocument(authorisation, claim, documentSelfPath);
+            @Override
+            public PDF getDocument() {
+                return document;
+            }
+        };
+    }
 
-            return document.getBytes();
-        }
+    private interface DocumentDetails {
```

Comment:
This interface and anonymous implementation for this is totally unnecessary, we need a data object here which is best achieved by using a `class`

Reply:
The intention is to avoid creating another class which is highly unlikely to be used outside the context of this class and hence this.